### PR TITLE
Add comments aa1

### PR DESCRIPTION
**Pull request by Theralun**
**Checklist:**

 
 - [ ] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no way to comment on posts.
- Currently likes are stored within the post documents in the database.
- Currently the like functionality does not operate properly from time to time.
- Currently the database is being read from multiple times within the code.
- Currently there is no `Dark Edition`.

**What is the new behavior (if this is a feature change)?**
- Comments and comment replies are now available to users.
- Likes are now stored in a seperate collection.
- Like functionality now updates like count locally for the user interface in addition to storing a like in the database.
- Database read accesses in normal site operation have been reduced.
- The `Dark Edition` has been added.

**Other information**:
- The version has been updated to `v1.5.0`.

### CODE REVIEW

File: src/store/actions/commentActions.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  };
+};
+
+// Load comments
+export const loadComments = () => {
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firestore = getFirestore();
+
+    // Apply where condition
+    const comments = firestore
```
Comment (Theralun) [inline src/store/actions/commentActions.js:172]: Let instead of const?

File: src/store/actions/commentActions.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+// Edit Comments
+export const editComment = (comment, newmsg) => {
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firebase = getFirebase();
+    const firestore = getFirestore();
+
+    // Get userId
+    let uid = comment.data.author.uid;
+    let loggedInUid = firebase.auth().currentUser.uid;
```
Comment (Theralun) [inline src/store/actions/commentActions.js:41]: `const`?

File: src/store/actions/likeActions.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  };
+};
+
+// Get amount of likes on a post
+export const getLikeAmount = (pid) => {
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firestore = getFirestore();
+    const firebase = getFirebase();
+
+    console.log("GET LIKE AMOUNT CALLED");
```
Comment (Theralun) [inline src/store/actions/likeActions.js:57]: `console.log()` Still needed?

File: src/store/actions/likeActions.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+// See if user has liked post already
+export const hasLiked = (pid) => {
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firestore = getFirestore();
+    const firebase = getFirebase();
+
+    // Get userId
+    const uid = firebase.auth().currentUser.uid;
+
+    const likes = firestore.collection("likes").doc(pid + uid);
```
Comment (Theralun) [inline src/store/actions/likeActions.js:84]: `const` or `let`?`

File: src/store/actions/likeActions.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firestore = getFirestore();
+    const firebase = getFirebase();
+
+    console.log("GET LIKE AMOUNT CALLED");
+
+    // Get userId
+    const uid = firebase.auth().currentUser.uid;
+
+    const likes = firestore.collection("likes").where("pid", "==", pid);
```
Comment (Theralun) [inline src/store/actions/likeActions.js:62]: `const` or `let`?

File: src/store/actions/userActions.js [hunk 6]
```diff
@@ -1,36 +1,64 @@
 //#region > Imports
-//> Additional
 // SHA265 algorithm
 import sha256 from "js-sha256";
 //#endregion
 
 //#region > Exports
```
Comment (Theralun) [inline src/store/actions/userActions.js:6]: Is this considered Exports or rather Functions?

File: src/store/reducers/commentReducer.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
+// Have initial state for when state is not ready to be passed
+const initState = {
+  commentError: null,
+};
+
+const commentReducer = (state = initState, action) => {
+  switch (action.type) {
+    case "EDITCOMMENT_SUCCESS":
+      console.log("Edit Success");
```
Comment (Theralun) [inline src/store/reducers/commentReducer.js:9]: Line break after.

File: src/store/reducers/commentReducer.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      return {
+        ...state,
+        commentError: null,
+      };
+    case "EDITCOMMENT_ERROR":
+      console.log("Edit error", action.err);
```
Comment (Theralun) [inline src/store/reducers/commentReducer.js:15]: Line break after.

File: src/store/reducers/commentReducer.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+      return {
+        ...state,
+        commentError: action.err,
+      };
+    case "CREATION_SPAM":
+      console.log("comment is spam", action.err);
```
Comment (Theralun) [inline src/store/reducers/commentReducer.js:21]: Line break after.

File: src/store/reducers/commentReducer.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+      return {
+        ...state,
+        commentError: action.err,
+      };
+    case "LOADCOMMENTS_SUCCESS":
+      console.log("Loading success");
```
Comment (Theralun) [inline src/store/reducers/commentReducer.js:27]: Line break after.

File: src/store/reducers/commentReducer.js [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+      return {
+        ...state,
+        authError: null,
+        results: action.results,
+        loading: false,
+      };
+    case "LOAD_LOADING":
+      console.log("Loading...");
```
Comment (Theralun) [inline src/store/reducers/commentReducer.js:35]: Line break after.

File: src/store/reducers/commentReducer.js [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+      return {
+        ...state,
+        loading: true,
+      };
+    case "LOAD_ERROR":
+      console.log("Loading error", action.err);
```
Comment (Theralun) [inline src/store/reducers/commentReducer.js:41]: Line break after.

File: src/store/reducers/commentReducer.js [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+      return {
+        ...state,
+        authError: action.err,
+        loading: false,
+      };
+    case "REMOVE_SUCCESS":
+      console.log("Remove success");
```
Comment (Theralun) [inline src/store/reducers/commentReducer.js:48]: Line break after.

File: src/store/reducers/commentReducer.js [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+      return {
+        ...state,
+      };
+    case "REMOVE_ERROR":
+      console.log("Remove error", action.err);
```
Comment (Theralun) [inline src/store/reducers/commentReducer.js:53]: Line break after.

File: src/store/reducers/commentReducer.js [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+      return {
+        ...state,
+      };
+    case "COMMENT_SUCCESS":
+      console.log("Comment created", action.commentId);
```
Comment (Theralun) [inline src/store/reducers/commentReducer.js:58]: Line break after.

File: src/store/reducers/commentReducer.js [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+      return {
+        ...state,
+      };
+    case "COMMENT_ERROR":
+      console.log("Comment not created", action.err);
```
Comment (Theralun) [inline src/store/reducers/commentReducer.js:63]: Line break after.

File: src/store/reducers/likeReducer.js [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+const initState = {
+  postError: null,
+  liked: false,
+  likecount: null,
+};
+
+const likeReducer = (state = initState, action) => {
+  switch (action.type) {
+    case "LIKECREATION_SUCCESS":
+      console.log("Creation success");
```
Comment (Theralun) [inline src/store/reducers/likeReducer.js:11]: Line break after.

File: src/store/reducers/likeReducer.js [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+      return {
+        ...state,
+        postError: null,
+      };
+    case "LIKECREATION_ERROR":
+      console.log("Creation error", action.err);
```
Comment (Theralun) [inline src/store/reducers/likeReducer.js:17]: Line break after.

File: src/store/reducers/likeReducer.js [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+      return {
+        ...state,
+        postError: action.err,
+      };
+    case "REMOVELIKE_SUCCESS":
+      console.log("Removal success");
```
Comment (Theralun) [inline src/store/reducers/likeReducer.js:23]: Line break after.

File: src/store/reducers/likeReducer.js [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+      return {
+        ...state,
+        postError: null,
+      };
+    case "REMOVELIKE_ERROR":
+      console.log("Removal error", action.err);
```
Comment (Theralun) [inline src/store/reducers/likeReducer.js:29]: Line break after.

File: src/store/reducers/likeReducer.js [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+      return {
+        ...state,
+        postError: action.err,
+      };
+    case "HASLIKED_SUCCESS":
+      console.log("hasliked success");
```
Comment (Theralun) [inline src/store/reducers/likeReducer.js:35]: Line break after.

File: src/store/reducers/likeReducer.js [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+      return {
+        ...state,
+        liked: action.liked,
+      };
+    case "HASLIKED_ERROR":
+      console.log("hasliked error", action.err);
```
Comment (Theralun) [inline src/store/reducers/likeReducer.js:41]: Line break after.

File: src/store/reducers/postReducer.js [hunk 23]
```diff
@@ -1,105 +1,77 @@
...
-      case 'COMMENT_ERROR':
-        console.log('Comment not created', action.err);
-        return {
-          ...state,
-        };
-      default:
-        return state;
+  switch (action.type) {
+    case "POSTCREATION_SUCCESS":
+      console.log("Creation success");
```
Comment (Theralun) [inline src/store/reducers/postReducer.js:9]: Line break after.

File: src/store/reducers/postReducer.js [hunk 24]
```diff
@@ -1,105 +1,77 @@
...
+      return {
+        ...state,
+        postError: null,
+      };
+    case "POSTCREATION_ERROR":
+      console.log("Creation error", action.err);
```
Comment (Theralun) [inline src/store/reducers/postReducer.js:15]: Line break after.

File: src/store/reducers/postReducer.js [hunk 25]
```diff
@@ -1,105 +1,77 @@
...
+      return {
+        ...state,
+        postError: action.err,
+      };
+    case "CREATION_SPAM":
+      console.log("Post is spam", action.err);
```
Comment (Theralun) [inline src/store/reducers/postReducer.js:21]: Line break after.

File: src/store/reducers/postReducer.js [hunk 26]
```diff
@@ -1,105 +1,77 @@
...
+      return {
+        ...state,
+        postError: action.err,
+      };
+    case "LOADPOSTS_SUCCESS":
+      console.log("Loading success");
```
Comment (Theralun) [inline src/store/reducers/postReducer.js:27]: Line break after.

File: src/store/reducers/postReducer.js [hunk 27]
```diff
@@ -1,105 +1,77 @@
...
+      return {
+        ...state,
+        authError: null,
+        results: action.results,
+        loading: false,
+      };
+    case "LOADPOSTS_LOADING":
+      console.log("Loading...");
```
Comment (Theralun) [inline src/store/reducers/postReducer.js:35]: Line break after.

File: src/store/reducers/postReducer.js [hunk 28]
```diff
@@ -1,105 +1,77 @@
...
+      return {
+        ...state,
+        loading: true,
+      };
+    case "LOADPOSTS_ERROR":
+      console.log("Loading error", action.err);
```
Comment (Theralun) [inline src/store/reducers/postReducer.js:41]: Line break after.

File: src/store/reducers/postReducer.js [hunk 29]
```diff
@@ -1,105 +1,77 @@
...
+      return {
+        ...state,
+        authError: action.err,
+        loading: false,
+      };
+    case "REMOVEPOSTS_SUCCESS":
+      console.log("Remove success");
```
Comment (Theralun) [inline src/store/reducers/postReducer.js:48]: Line break after.

File: src/store/reducers/postReducer.js [hunk 30]
```diff
@@ -1,105 +1,77 @@
...
+      return {
+        ...state,
+      };
+    case "REMOVEPOSTS_ERROR":
+      console.log("Remove error", action.err);
```
Comment (Theralun) [inline src/store/reducers/postReducer.js:53]: Line break after.

File: src/store/reducers/postReducer.js [hunk 31]
```diff
@@ -1,105 +1,77 @@
...
+      return {
+        ...state,
+      };
+    case "COMMENT_SUCCESS":
+      console.log("Comment created", action.postId);
```
Comment (Theralun) [inline src/store/reducers/postReducer.js:58]: Line break after.

File: src/store/reducers/postReducer.js [hunk 32]
```diff
@@ -1,105 +1,77 @@
...
+      return {
+        ...state,
+      };
+    case "COMMENT_ERROR":
+      console.log("Comment not created", action.err);
```
Comment (Theralun) [inline src/store/reducers/postReducer.js:63]: Line break after.

File: src/store/actions/commentActions.js (hunk 1 above)
Comment (Aichnerc) [reply src/store/actions/commentActions.js:172]: If a value is not requird to change, you should use const.

File: src/store/actions/commentActions.js (hunk 2 above)
Comment (Aichnerc) [reply src/store/actions/commentActions.js:41]: Absolutely.

File: src/store/actions/likeActions.js (hunk 4 above)
Comment (Aichnerc) [reply src/store/actions/likeActions.js:84]: If a value is not requird to change, you should use const.

File: src/components/pages/ProfilePage/index.jsx [hunk 33]
```diff
@@ -451,10 +473,11 @@ class ProfilePage extends React.Component {
   };
 
   render() {
-    const { auth, profile } = this.props;
-
+    const { auth, profile, comments } = this.props;
     if (auth.uid === undefined) return <Redirect to="/login" />;
```
Comment (LucaAllmaier) [inline src/components/pages/ProfilePage/index.jsx:477]: I think there should be a new line

File: src/store/actions/likeActions.js (hunk 5 above)
Comment (Aichnerc) [reply src/store/actions/likeActions.js:62]: If a value is not requird to change, you should use const.

File: src/components/pages/ProfilePage/index.jsx [hunk 34]
```diff
@@ -451,10 +473,11 @@ class ProfilePage extends React.Component {
   };
 
   render() {
-    const { auth, profile } = this.props;
-
+    const { auth, profile, comments } = this.props;
```
Comment (LucaAllmaier) [inline src/components/pages/ProfilePage/index.jsx:476]: I think there should be new line

File: src/store/actions/userActions.js (hunk 6 above)
Comment (Aichnerc) [reply src/store/actions/userActions.js:6]: Functions fit it better, yes.

File: src/components/molecules/Comment/index.jsx [hunk 35]
```diff
@@ -0,0 +1,592 @@
...
+      likeCount: this.state.likeCount + amount,
+    });
+  };
+
+  removeComment = (comment) => {
+    this.props.removeComment(comment);
+    this.props.refreshData();
+  };
+
+  _calculateTimeAgo = (timestamp) => {
```
Comment (LucaAllmaier) [inline src/components/molecules/Comment/index.jsx:153]: functions no longer start with underscores.

File: src/components/molecules/Post/index.jsx [hunk 36]
```diff
@@ -0,0 +1,484 @@
...
+                        </div>
+                        <div>Receiving current status</div>
+                      </MDBPopoverBody>
+                    )}
+                  </div>
+                </MDBPopover>
+                {post.data.details.feeling && (
+                  <i className="feeling">
+                    <MDBIcon
+                      icon={post.data.details.feeling.icon}
```
Comment (LucaAllmaier) [inline src/components/molecules/Post/index.jsx:284]: Hufsky would probably hate that object-deepness^^

File: src/store/actions/commentActions.js [hunk 37]
```diff
@@ -0,0 +1,199 @@
...
+        msg: newComment.comment,
+        author: newComment.author,
+        timestamp: timestamp,
+        pid: newComment.pid,
+        cid: newComment.cid,
+        visible: true,
+      })
+      .then(() => {
+        dispatch({ type: "CREATION_SUCCESS", newComment });
+        return;
```
Comment (Aichnerc) [inline src/store/actions/commentActions.js:25]: Return useless.

File: src/store/actions/commentActions.js [hunk 38]
```diff
@@ -0,0 +1,199 @@
...
+      .set(
+        {
+          uid: uid,
+          likes: localLikes,
+        },
+        { merge: true }
+      )
+      .then(() => {
+        dispatch({ type: "LIKE_SUCCESS", id });
+        return;
```
Comment (Aichnerc) [inline src/store/actions/commentActions.js:100]: Return useless.

File: src/store/actions/commentActions.js [hunk 39]
```diff
@@ -0,0 +1,199 @@
...
+    // Create comment
+    firestore
+      .collection("likes")
+      .doc(id)
+      .update({
+        likes: likes,
+      })
+      .then(() => {
+        dispatch({ type: "UNLIKE_SUCCESS", id });
+        return;
```
Comment (Aichnerc) [inline src/store/actions/commentActions.js:126]: Return useless.

**Pull request merged**